src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const arr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        newState = {};
+        break;
+
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case `removeProperties`:
+        for (const type of action.keysToRemove) {
+          delete newState[type];
+        }
```

Comment:
add default case
https://stackoverflow.com/questions/4649423/should-switch-statements-always-contain-a-default-clause